publisher.rb

```diff
@@ -1,25 +1,46 @@
 require 'aws-sdk-sqs'
 require_relative 'load_aws'
 require_relative 'aws_sqs_client'
+require_relative 'q_it_errors'
 
 class Publisher
   
   attr_reader :sqs, :sleep_period, :queue_name
 
   def initialize(options)
     @sqs = AwsSQSClient.new.client
-    @sleep_period = valid_sleep_period?(options[1]) ? options[1].to_i : 5
-    @queue_name = options[0]
+    @queue_name = valid_queue_name(options[0])
+    @sleep_period = validate_sleep_period(options[1])
   end
 
   # For Standard Queue
   def create_std_q
     sqs.create_queue(queue_name: queue_name)
   end
 
-  def valid_sleep_period?(sleep_period)
+  def validate_sleep_period(sleep_period)
     max_sleep_period = /\A([5-9]|1\d{1}|20)\z/
-    !sleep_period.nil? && sleep_period.match(max_sleep_period)
+    begin
+      raise QItNullSleepTimeError, "QItNullSleepTimeError :: #{self} Sleep period cannot be nil." if sleep_period.nil?
+      raise QItArgumentError, "QItArgumentError :: #{self} Invalid sleep period #{sleep_period}. Sleep period range is between 5-20 seconds." unless sleep_period.match(max_sleep_period)
+      sleep_period
+    rescue QItArgumentError => e
```

Comment:
Do we want to have a `rescue` here? We only seem to be printing it out and exiting. For here and L30.
Just raise - we should not care about rescuing at least for this code as we need that the program should break due to incorrect input.